### PR TITLE
Fix contributor workflow

### PR DESCRIPTION
**Pull request by casperklein**
# Description

This should finally fix the contributors workflow. I've tested it successfully in a test repo.

Fixes #2224

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/contributors.yml [hunk 1]
```diff
@@ -46,7 +46,10 @@ jobs:
       # The lint workflow is not important for this type of PR, skip it and pretend it was successful:
       - name: 'Get the latest commit hash from the contributors-update branch'
         id: commit-data
-        run: 'echo "::set-output name=head_sha::$(git rev-parse contributors-update)"'
+        run: |
+          git pull
+          git checkout contributors-update
+          echo "::set-output name=head_sha::$(git rev-parse contributors-update)"
```
Comment (polarathene) [inline .github/workflows/contributors.yml:52]: The relevant commit should have been committed locally to be pushed to the remote `contributors-update` branch. I believe `git rev-parse @` would work just as well here, without needing to pull.

I was writing a response to you with that suggested fix, but forgot to send it prior to working on a PR that implemented it.

**Review by polarathene — APPROVED** [review]

**Comment by polarathene** [comment]
I don't mind this PR being merged, but we may want to consider the one [I submitted roughly at the same time](https://github.com/docker-mailserver/docker-mailserver/pull/2227), although it's got more going on..

File: .github/workflows/contributors.yml (hunk 1 above)
Comment (casperklein) [reply .github/workflows/contributors.yml:52]: I tried that before. Without `git pull`, `git checkout contributors-update` fails.

**Review by wernerfred — APPROVED** [review]

**Review by georglauterbach — APPROVED** [review]

File: .github/workflows/contributors.yml (hunk 1 above)
Comment (polarathene) [reply .github/workflows/contributors.yml:52]: > Not removing `git pull `

Doesn't the PyGithub API used by the `add-contributors` action not commit locally to push to the remote branch? It seems odd that a `git pull` would be required if the commit pushed is the same commit SHA locally and remote?

Something else perhaps is going on that I missed/misunderstood?

Comment (casperklein) [reply .github/workflows/contributors.yml:52]: I cannot help you with that. I don't know how `add-contributors` action works in detail (not very familiar with python).

**Pull request merged**